packages/logic-utils/src/utils/documentation.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  UI5DeprecatedInfo,
+  UI5SemanticModel,
+} from "@ui5-language-assistant/semantic-model-types";
+import { ReplaceFunction, forEach, unescape } from "lodash";
+
+const MAX_SNIPPET_LENGTH = 500;
+
+export function getDeprecationPlainTextSnippet(
+  title: string | undefined,
+  deprecatedInfo: UI5DeprecatedInfo,
+  model: UI5SemanticModel
+): string {
+  let text = deprecatedInfo.text;
+  if (text !== undefined) {
+    text = convertJSDocToPlainText(text, model);
+    // Only take the first sentence.
+    // A sentence is defined by a dot followed by a whitespace character.
+    // TODO Should we allow the result to have more than 1 line? (If so, add "s" flag to the regex)
+    const matches = text.match(/^((?:.*?)\.)\s/);
+    if (matches !== null) {
+      text = matches[1];
+    }
+    // Limit the length of the snippet
+    if (text.length > MAX_SNIPPET_LENGTH) {
+      text = text.substring(0, MAX_SNIPPET_LENGTH) + "...";
+    }
+  }
+
+  const doc = getDeprecationMessage({
+    title,
+    since: deprecatedInfo.since,
+    text,
+  });
+  return doc;
+}
+
+export function getDeprecationMessage({
+  title,
+  since,
+  text,
+}: {
+  title?: string;
+  since: string | undefined;
```

Comment:
if we mark the type as optional `since?` would passing undefined to it raise a compilation error?

Reply:
No

Reply:
Should we have a more consistent modeling of optional props here?
Or is this constrained by what we actually get from the model?

Maybe it does not matter enough?